### ISSUE TITLE
Got error: The controller for URI "/sell/orders/" is not callable. Class "Skaparate\Controller\Admin\CustomOrdersController" does not exist.

### ISSUE THREAD

**Issue by Bugfunder** [skaparate/prestashop-sample-module#1]
Any idea how to resolve / debug it?

**Comment by skaparate** [comment]
Hello!

If you cloned the project, instead of using the release, Did you run `composer install` first? It's required so the autoloader is generated (there needs to be a vendor folder on the project).

**Comment by Bugfunder** [comment]
No, did not run it (was not written in your readme). In which folder should
I run it?

вт, 29 черв. 2021 о 14:40 skaparate ***@***.***> пише:

**Comment by skaparate** [comment]
If you cloned the repository, then in the root of the module, under `PrestaShop/modules/sk_test`:

```
cd path_to_prestashop/modules/sk_test
composer install
```
If you used the release zip, then it should work out of the box.

**Comment by Bugfunder** [comment]
Yes, I installed it and got it writing Hello World.

Can you point out where your release zip is?

Because in my case

composer install

generated lots of files in vendor folder, and I wonder if they are all
needed.



вт, 29 черв. 2021 о 14:48 skaparate ***@***.***> пише:

**Comment by skaparate** [comment]
It's [under the releases of the repository](https://github.com/skaparate/prestashop-sample-module/releases).

To understand what composer does, please [refer to the documentation](https://getcomposer.org/doc/00-intro.md).

**Comment by skaparate** [comment]
By the way, the `"require-dev"` of the composer file are just utilities to help you develop the module and are not really required (you may replace or remove them)

**Comment by Bugfunder** [comment]
Thanks, in my case composer install generates lots of folders in vendor:

autoload.php  bin  composer  doctrine  friendsofphp  myclabs  nikic
 phar-io  php-cs-fixer  phpdocumentor  phpspec  phpunit  prestashop  psr
 sebastian  squizlabs  symfony  theseer  webmozart

Perhaps it has to do with my installation of composer. Will give a try to
your release zip instead.




вт, 29 черв. 2021 о 21:45 skaparate ***@***.***> пише:

**Comment by Bugfunder** [comment]
Now I get a different problem. When I go to Orders/Orders for the first
time, I get " Hello world!
", as intended.

When I do it for the next time, I get
Page not found
The controller AdminOrders is missing or invalid.
So apparently it tries to load the old (legacy) controller.


ср, 30 черв. 2021 о 09:31 Yarick ***@***.***> пише:

**Comment by skaparate** [comment]
> Thanks, in my case composer install generates lots of folders in vendor: autoload.php bin composer doctrine friendsofphp myclabs nikic phar-io php-cs-fixer phpdocumentor phpspec phpunit prestashop psr sebastian squizlabs symfony theseer webmozart

Don't worry, that's OK. Those are the dependencies (and the dependencies between) used on the `"require-dev"` of the `composer.json` file. Remove all the `"required-dev"` if you like.

> When I do it for the next time, I get
Page not found
The controller AdminOrders is missing or invalid.
So apparently it tries to load the old (legacy) controller.

That's something I didn't test for :stuck_out_tongue:. If you can wait, I'll review it in a couple of weeks (I have another job right now), otherwise you will have to try it on your own, sorry :sweat:. Regardless, I thank you for letting me know :smiley:.

**Comment by Bugfunder** [comment]
Yes, I can wait, no problem. Currently put my controller under
PrestashopBundle.

BTW, in the official documentation there is nothing about composer stuff.

ср, 30 черв. 2021 о 18:15 skaparate ***@***.***> пише:

**Comment by skaparate** [comment]
> Currently put my controller under PrestashopBundle.

You mean you put it directly under `path_to_prestashop/src/PrestaShopBundle`? If so, there's a chance that when you update PrestaShop your controller will be replaced/removed. 

> BTW, in the official documentation there is nothing about composer stuff

It's mentioned here: https://devdocs.prestashop.com/1.7/modules/concepts/composer/#composer

Remember that this is just a sample and not meant to serve as a rule for everyone. If you don't want to use composer on your project, switch it to another dependency manager or remove it (if you know what you're doing) 😃.

**Comment by Bugfunder** [comment]
I just wish it works without any dependency manager other than the standard
one used in Prestashop. The twig template works without it,
and also the page

https://devdocs.prestashop.com/1.7/modules/concepts/controllers/admin-controllers/override-decorate-controller/

does not mention the need to use composer.

As I do updates via git, I do not think they can remove my controller.


ср, 30 черв. 2021 о 22:36 skaparate ***@***.***> пише:

**Comment by skaparate** [comment]
> I just wish it works without any dependency manager other than the standard
one used in Prestashop. The twig template works without it,
and also the page
>
> https://devdocs.prestashop.com/1.7/modules/concepts/controllers/admin-controllers/override-decorate-controller/
> 
> does not mention the need to use composer.

Like I said, you can simply remove it and adapt it however you like :)

**Comment by Bugfunder** [comment]
Yeah, but the problem is that it does not work without..

чт, 1 лип. 2021 о 00:07 skaparate ***@***.***> пише:
